src/txn/txn_timestamp.c

```diff
@@ -946,6 +946,32 @@ __wt_txn_set_timestamp(WT_SESSION_IMPL *session, const char *cfg[])
     return (0);
 }
 
+/*
+ * __wt_txn_set_timestamp_commit --
+ *     Directly set the commit timestamp in a transaction, bypassing parsing logic. Prefer this to
+ *     @ref __wt_txn_set_timestamp when string parsing is a performance bottleneck.
+ */
+int
+__wt_txn_set_timestamp_commit(WT_SESSION_IMPL *session, wt_timestamp_t ts)
+{
+    WT_CONNECTION_IMPL *conn;
+    WT_DECL_RET;
+
+    WT_TRET(__wt_txn_context_check(session, true));
+
+    if (ts != WT_TS_NONE)
+        WT_RET(__wt_txn_set_commit_timestamp(session, ts));
+
+    __wt_txn_publish_durable_timestamp(session);
```

Comment:
Something doesn't look right to me. If we don't have any timestamp to set (WT_TS_NONE as an argument), why do we publish durable timestamp. I know it follows from ```__wt_txn_set_timestamp```, but it looks off.

Reply:
@kommiharibabu , any idea around here ?

Reply:
The `__wt_txn_publish_durable_timestamp` doesn't do anything if commit or durable timestamp is not set. While going through this function, it doesn't have all the validation checks that are done in other code flow.

Reply:
`__wt_txn_publish_durable_timestamp` seems to do stuff when the commit timestamp is set, so I thought it appropriate to leave this call here. I'd be lying if I said I completely understood why it needs to act on the commit timestamp though.

Reply:
Right - the name of the function is misleading. It assumes that everyone knows that a commit timestamp implicitly becomes the durable timestamp if a different one isn't set.
I think it could be renamed to `wt_txn_publish_user_timestamp`, and its intent would be clearer. It publishes any timestamp for a transaction that needs to be published.